metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py

```diff
@@ -251,14 +251,20 @@ class DBTCommonConfig(StatefulIngestionConfigBase, LineageConfig):
     )
     backcompat_skip_source_on_lineage_edge: bool = Field(
         False,
-        description="Prior to version 0.8.41, lineage edges to sources were directed to the target platform node rather than the dbt source node. This contradicted the established pattern for other lineage edges to point to upstream dbt nodes. To revert lineage logic to this legacy approach, set this flag to true.",
+        description="[deprecated] Prior to version 0.8.41, lineage edges to sources were directed to the target platform node rather than the dbt source node. This contradicted the established pattern for other lineage edges to point to upstream dbt nodes. To revert lineage logic to this legacy approach, set this flag to true.",
     )
 
     incremental_lineage: bool = Field(
         # Copied from LineageConfig, and changed the default.
         default=False,
         description="When enabled, emits lineage as incremental to existing lineage already in DataHub. When disabled, re-states lineage on each run.",
     )
+    backcompat_no_env_in_assertion_guid: bool = Field(
```

Comment:
Should we call this "backcompat"? This feels a bit weird. What is the case in which people would want to keep the previous behavior? In my opinion previous behavior will ALWAYS be wrong.

Reply:
If someone was previously only running dbt dev ingestion (no prod), this would cause all of their assertion URNs to change.

Reply:
we can call it legacy or something else

Reply:
It's actually probably okay. They would just need to rerun the assertions / ingestion and it would get correctly attached going forward. So the impact isn't super terrible I think...

Let's use legacy or things like this... But this is the type of config that most users should never need to see nor care about

Reply:
Don't forget to add this to deprecation list https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md

Reply:
@anshbansal we've changed it so that this is only opt-in